File: Aplikacija/Front deo/erestoranFront/src/app/feature/restaurants/models/Restoran.ts
import { Jela } from "./Jela";
import { Komentar } from "./Komentar";
import { Ocena } from "./Ocena";
import { Sala } from "./Sala";

export interface Restoran
{
    id: number;
    naziv: string;
    opis: string;
    adresa: string;
    slika: string;
    meni: Jela[];
    komentari: Komentar[];
    sale: Sala[];
    ocene: Ocena[];
}

File: Aplikacija/Back deo/ERestoran/Models/Restoran.cs
namespace Models
{
    public class Restoran
    {
        [Key]
        public int ID { get; set; }
        public required string Naziv { get; set; }
        public required string Adresa { get; set; }
        public string? Opis { get; set; }
        public string? Slika { get; set; }
        public List<Sala>? Sale { get; set; } = new List<Sala>();
        public List<Hrana>? Meni { get; set; } = new List<Hrana>();
        public List<OcenaRestorana>? Ocene { get; set; } = new List<OcenaRestorana>();
        public List<KomentarRestorana>? Komentari { get; set; } = new List<KomentarRestorana>();
    }
}